File: src/main/java/com/Project1/config/CassandraConfiguration.java
package com.Project1.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.cassandra.config.AbstractCassandraConfiguration;
import org.springframework.data.cassandra.config.SchemaAction;

/**
 * properties for Cassandra Configuration
 */
@Configuration
public class CassandraConfiguration extends AbstractCassandraConfiguration {

    @Value("${spring.data.cassandra.keyspace-name}")
    private String keySpace;

    @Value("${spring.data.cassandra.contact-points}")
    private String contactPoints;

    @Value("${spring.data.cassandra.port}")
    private int port;

    @Value("${spring.data.cassandra.schema-action}")
    private String schemaAction;

    public String getKeySpace() {
        return keySpace;
    }

    public void setKeySpace(String keySpace) {
        this.keySpace = keySpace;
    }

    @Override
    protected String getKeyspaceName() {
        return keySpace;
    }

    @Override
    public String getContactPoints() {
        return contactPoints;
    }

    public void setContactPoints(String contactPoints) {
        this.contactPoints = contactPoints;
    }

    @Override
    public int getPort() {
        return port;
    }

    public void setPort(int port) {
        this.port = port;
    }

    @Override
    public SchemaAction getSchemaAction() {
        return SchemaAction.CREATE_IF_NOT_EXISTS;
    }
}

File: src/main/kotlin/org/ProjectV1/config/CassandraConfiguration.kt
package org.ProjectV1.config

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Configuration
import org.springframework.data.cassandra.config.AbstractCassandraConfiguration
import org.springframework.data.cassandra.config.SchemaAction

/**
 * properties for Cassandra Configuration
 */
@Configuration
class CassandraConfiguration : AbstractCassandraConfiguration() {
    @Value("\${spring.data.cassandra.keyspace-name}")
    private var keySpace: String = ""

    @Value("\${spring.data.cassandra.contact-points}")
    private var contactPoints: String = ""

    @Value("\${spring.data.cassandra.port}")
    private var port = 0

    @Value("{spring.data.cassandra.schema-action}")
    private var schemaAction: String = ""

    override fun getKeyspaceName(): String {
        return keySpace
    }

    public override fun getContactPoints(): String {
        return contactPoints
    }

    fun setContactPoints(contactPoints: String) {
        this.contactPoints = contactPoints
    }

    public override fun getPort(): Int {
        return port
    }

    fun setPort(port: Int) {
        this.port = port
    }

    override fun getSchemaAction(): SchemaAction {
        return SchemaAction.CREATE_IF_NOT_EXISTS
    }
}